### PR TITLE
17822 Update business response to return alternateNames list

### PR DESCRIPTION
**Pull request by argush3**
*Issue #:* /bcgov/entity#17822

*Description of changes:*

* Removed `businessName` from GET /business response
* Added `alternateNames` list to GET /business response.  Note that this will support the use case of businesses such as BC LTD businesses own multiple firms and as such have multiple operating names. 
* Added to test `alternateNames` property on `LegalEntity` model.  
* Updated existing tests

**_SP alternate names example_**
```http response
GET http://127.0.0.1:5000/api/v2/businesses/FM0554669

{
  "business": {
    "identifier": "FM0554669",
    "legalName": "DAYTON ALLAN SCHUNTER",
    "legalType": "SP",
    "alternateNames": [
      {
        "entityType": "SP",
        "identifier": "FM0554669",
        "nameRegisteredDate": "2011-04-14T22:50:14+00:00",
        "nameStartDate": "2011-04-14",
        "operatingName": "BOOTS LOWBED LOG HAUL - IMPORT_TEST"
      }
    ],
    ...
  }
}
```

**_GP alternate names example_**
```http response
GET http://127.0.0.1:5000/api/v2/businesses/FM0116406

{
  "business": {
    "identifier": "FM0116406",
    "legalName": "400396 BC LTD, 657719 BC LTD., et al",
    "legalType": "GP",
    "alternateNames": [
      {
        "entityType": "GP",
        "identifier": "FM0116406",
        "nameRegisteredDate": "1991-03-06T08:00:00+00:00",
        "nameStartDate": "1991-03-06",
        "operatingName": "EXCELL BATTERY COMPANY - IMPORT_TEST"
      }
    ],
    ...
  }
}
```

**_No alternate names example_**
```http response
GET http://127.0.0.1:5000/api/v2/businesses/CP0001199

{
  "business": {
    "identifier": "CP0001199",
    "legalName": "LE COEUR HOUSING CO-OPERATIVE",
    "legalType": "CP",
    ...
  }
}

```

**_Non-firm with multiple alternate names example_**
```http response
GET http://127.0.0.1:5000/api/v2/businesses/BC1230010

{
  "business": {
    "identifier": "BC1230010",
    "legalName": "BC1230010",
    "legalType": "BEN",
    "alternateNames": [
      {
        "entityType": "SP",
        "identifier": "FM1032630",
        "nameRegisteredDate": "2023-03-16T21:35:43.599246+00:00",
        "nameStartDate": "2023-03-16",
        "operatingName": "FIX"
      },
      {
        "entityType": "SP",
        "identifier": "FM1032736",
        "nameRegisteredDate": "2023-03-17T23:39:25.980718+00:00",
        "nameStartDate": "2023-03-17",
        "operatingName": "SOME DBA TEST"
      },
      {
        "entityType": "GP",
        "identifier": "FM1030515",
        "nameRegisteredDate": "2023-02-22T22:41:03.306935+00:00",
        "nameStartDate": "2023-02-22",
        "operatingName": "AMIGOS"
      }
    ],
    ...
  }
}
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/models/alternate_name.py [hunk 1]
```diff
@@ -55,7 +55,11 @@ class NameType(BaseEnum):
     legal_entity_id = db.Column('legal_entity_id', db.Integer, db.ForeignKey('legal_entities.id'))
     change_filing_id = db.Column('change_filing_id', db.Integer, db.ForeignKey('filings.id'), index=True)
 
+    # relationships
+    legal_entity = db.relationship("LegalEntity", back_populates="_alternate_names")
```
Comment (argush3) [inline legal-api/src/legal_api/models/alternate_name.py:59]: This was required as sqlalchemy was throwing warnings.

**Review by kzdev420 — APPROVED** [review]
Looks good to me.

File: legal-api/src/legal_api/models/legal_entity.py [hunk 2]
```diff
@@ -479,11 +479,43 @@ def business_name(self):
...
+    @property
+    def alternate_names(self):
+        """Return operating names for a business if any."""
+        le_alias = aliased(LegalEntity)
+        alternate_names = (
+            db.session.query(AlternateName.identifier,
+                             AlternateName.name,
+                             AlternateName.start_date,
+                             le_alias.entity_type,
+                             le_alias.founding_date)
```
Comment (thorwolpert) [inline legal-api/src/legal_api/models/legal_entity.py:496]: Is end_date not included, as we're deleting the AlternateName when it is ended?

**Review by thorwolpert — COMMENTED** [review]
Looks like we covered in _chat_, but outstanding was if we're deleting the name when the end_date is set.

Can we also push the model changes to 
https://github.com/bcgov/lear/tree/dev_legal_name_changes/python/common/business-registry-model

File: legal-api/src/legal_api/models/legal_entity.py (hunk 2 above)
Comment (argush3) [reply legal-api/src/legal_api/models/legal_entity.py:496]: This is how I've converted the existing data right now.  

image

I think maybe each of these history rows should have a unique id and two corresponding history entries.  One with the initial entry and the end-dated entry.  And the end-dated entries would not be in `alternate_names`.  If we want this, I'll need to make some updates to the conversion scripts.

Both approaches would mean that `alternate_names` will only contain alternate name entries that were not end-dated.

Comment (thorwolpert) [reply legal-api/src/legal_api/models/legal_entity.py:494]: Cool, pretty sure that's what we agreed to, and is consistent with the rest of the history table.
I like it!

**Review by thorwolpert — APPROVED** [review]
I like this.
With this change, do you think we're ready to publish the next version of the OAS doc?

**Review by vysakh-menon-aot — APPROVED** [review]

**Comment by argush3** [comment]
> I like this. With this change, do you think we're ready to publish the next version of the OAS doc?

I think we are close.  The only thing is we haven't really tested everything work together with the FE.  I guess we can just update the doc again if we need to make any tweaks.

Also, business-registry-model-updates have been pushed.

**Pull request merged**